### ISSUE TITLE
[GDScript 2.0] [regression] parse error on statement-less file 

### ISSUE THREAD

**Issue by Scony** [godotengine/godot#52028]
### Godot version

master #fb176d5f6e6dc2b43c630a041a8ab7d737173ec4

### System information

manjaro linux

### Issue description

While empty (0 bytes) file will get parsed successfully, the following files will not (they did in Godot `3.x`):

- newline, newlines:

```


```

 - comment + newline

```
# asd

```

Error:
```
SCRIPT ERROR: Parse Error: Unexpected "Newline" in class body.
```

### Steps to reproduce

compile

**Comment by lyuma** [comment]
I think it is acceptable for empty files to produce an error.

Empty files can be indicative of problems the user may wish to know about, such as files being overwritten or cleared. Is there any usecase in which having an empty gdscript file (without `extends`) is useful?

If an empty class is desired (for example, a common superclass with no behavior), it is good practice to declare the GDScript as `extends RefCounted` or `extends Node`, etc.

**Comment by Calinou** [comment]
Most scripting languages don't produce an error when running an empty file, but I agree it makes sense to produce an error in Godot's context. That said, the error message should be changed to be more helpful (e.g. `"Missing `extends <class>` declaration."`).

**Comment by Scony** [comment]
@lyuma scripts extend `Object` (or `Reference` - I'm  not sure) by default. I don't see any point of writing such statement in case of e.g. utility scripts where one has only e.g. static functions. To me, putting misleading `extends` in such circumstances ishould be considered a code smell.

Moreover - what about situation where someone, for whatever reason selects all the code in the file and turns it into comments?
I don't think some newbies will know they have to put statement like `extends` or `pass` to make it working - while such situation may be quite common. Imagine a newbie that is frustrated due to code not working - if he must finish his session, he will select all, comment all and save file with 3 shortcuts. I don't see a reason why such person's project should fail on file full of newlines and comments.

**Comment by lyuma** [comment]
Ok, I think I understand the original issue a bit better. I missed the fact that entirely empty scripts are allowed even in current master.

So the issue is that even though entirely empty scripts are currently allowed, for some reason in Godot master, adding comments and newlines to empty scripts is not allowed. So there is clearly some inconsistency, and thus the issue isn't about scripts without 'extends`.

(also of note, adding an empty annotation like `@tool` will cause the otherwise-empty script with comments to be accepted, even though it shouldn't have any effect.)

Furthermore, although I've never written GDScript without it, you seem to be correct that the `extends` line is not required. In fact, `extends` seems to be utterly ignored on scripts assigned through the Script field. (For example, Godot seems to allow a script which says `extends Node` to be assigned as the script on a `Resource`.)

In short, I agree with you that there's clearly some bug here. I also get the usecase of being allowed to comment out everything in the script.
